dut_power: make GPIOs open drain

hnez opened the pull request:
This PR improves the EMI performance of the TAC (at least when measuring the emissions on pins of the IOBus port in isolation).

How can that be you ask? The `VCC_IO` supply line powering all of the SoC's GPIO pins is necessarily affected by high frequency switching of e.g. the RGMII and eMMC lines. This results in GPIOs that are actively driven high also containing said noise.
This PR makes it so that the lines are never driven high by the SoC, relying instead on pull-up resistors on the power board.

But the `DUT_PWR_EN` and `DUT_PWR_DISCH` don't leave the device, they only go to the flat flex connector to the power board, I hear you ask. How can they couple out of the IOBus connector?
It looks like the noise couples onto the IOBus signal lines that are right next to the GPIOs on the flat flex connector.

Fun times were had figuring that out.

KarlK90 commented on src/dut_power.rs:
```diff
@@ -286,17 +286,23 @@ impl DutPwrThread {
-            DISCHARGE_LINE_ASSERTED,
-            "tacd",
-        )?;
+        let pwr_line = find_line("DUT_PWR_EN")
+            .ok_or_else(|| anyhow!("Could not find GPIO line DUT_PWR_EN"))?;
+        let discharge_line = find_line("DUT_PWR_DISCH")
+            .ok_or_else(|| anyhow!("Could not find GPIO line DUT_PWR_DISCH"))?;
+
+        // Early TACs have a i2c port expander on the power board.
+        // This port expander is output only and can not be configured as
+        // open drain.
+        // The outputs on later TACs should however be driven open drain
+        // for EMI reasons.
+        let flags = match pwr_line.chip().label() {
+            "pca9570" => LineRequestFlags::OUTPUT,
+            _ => LineRequestFlags::OUTPUT | LineRequestFlags::OPEN_DRAIN,
+        };
+
+        let pwr_line = pwr_line.request(flags, 1 - PWR_LINE_ASSERTED, "tacd")?;
+        let discharge_line = discharge_line.request(flags, DISCHARGE_LINE_ASSERTED, "tacd")?;
```
Is this a critical error if the gpio lines can't be found here? Because the `tacd` will panic as the result is unwraped. If no we could just log an error condition here?

hnez replied:
> Is this a critical error if the gpio lines can't be found here?

Yes indeed. All TACs come equipped with a power board, so if we can find it there is likely something wrong with e.g. the current devicetree and this RAUC boot slot should be marked bad.
The idea is to let the `tacd` startup fail in all cases that indicate such an error to make sure that the RAUC slot will be eventually marked bad.

hnez commented:
The merge of #29 resulted in merge conflicts in `src/digital_io/gpio/demo_mode.rs` and `src/digital_io/gpio/test.rs` as it removed the need to handle GPIO inputs at all.
I've rebased this branch on the current `main` and fixed the merge conflicts.
Please re-aprove and merge at will.
